storage/storagepb/convert.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storagepb
+
+import (
+	"fmt"
+
+	"github.com/google/trillian/merkle/smt"
+	"github.com/google/trillian/storage/tree"
+)
+
+// Unmarshal converts the given SubtreeProto to a Merkle tree tile.
+func Unmarshal(sp *SubtreeProto) (smt.Tile, error) {
+	if d := sp.Depth; d <= 0 {
+		return smt.Tile{}, fmt.Errorf("wrong depth %d, want > 0", d)
+	}
+	height := uint(sp.Depth)
```

Comment:
All proto fields should be accessed via the `Get*` functions for API compatibility with future proto changes.